### PR TITLE
f/linear-ysi-interpolation

### PR DESCRIPTION
**Pull request by jason-curtis**
[asana ticket](https://app.asana.com/0/819671808102776/1122858111406195/f)

Linearly interpolate YSI data when timestamps don't perfectly match.

## Manual testing:

used this yesterday on real data and got a better fit.

## TODO:

* regenerate documentation
* update snippet & latest experiment notebooks

### CODE REVIEW

File: osmo_jupyter/ysi_test.py [hunk 1]
```diff
@@ -68,45 +68,86 @@ def test_prefixes_with_YSI(self):
...
+                'ysi seconds match other seconds',
+                None,
+                [3, 5, 7],
+                [3, 5, 7],
+                [3, 5, 7],
+                [3, 5, 7]
+            ),
+            (
+                'interpolates to match data with default method',
+                None,
```
Comment (EvanSimpson) [inline osmo_jupyter/ysi_test.py:97]: Is it worth explicitly testing using `'slinear'` as a parameter?

**Review by EvanSimpson — APPROVED** [review]

File: osmo_jupyter/ysi.py [hunk 2]
```diff
@@ -53,6 +54,9 @@ def join_nearest_ysi_data(
                 parse_dates=['Timestamp']
             ).set_index('Timestamp')
         other_data_timestamp_column: Default: 'timestamp'. Column name in other_data containing timestamps.
+        interpolation_method: Default: 'slinear' method used when interpolating YSI data.
```
Comment (jaimemarijke) [inline osmo_jupyter/ysi.py]:
```suggestion
        interpolation_method: Default: 'slinear'. Method used when interpolating YSI data.
```

File: osmo_jupyter/ysi_test.py [hunk 3]
```diff
@@ -68,45 +68,86 @@ def test_prefixes_with_YSI(self):
...
+            (
+                'ysi seconds match other seconds',
+                None,
+                [3, 5, 7],
+                [3, 5, 7],
+                [3, 5, 7],
+                [3, 5, 7]
+            ),
+            (
+                'interpolates to match data with default method',
```
Comment (jaimemarijke) [inline osmo_jupyter/ysi_test.py]:
```suggestion
                'interpolates to match data with default (slinear) method',
```

File: osmo_jupyter/ysi_test.py [hunk 4]
```diff
@@ -68,45 +68,86 @@ def test_prefixes_with_YSI(self):
...
         self,
         name,
+        interpolation_method,
         other_seconds,
         ysi_seconds,
         expected_other_seconds,
         expected_ysi_seconds
     ):
-        actual = module.join_nearest_ysi_data(
+        # Only override the "method" kwarg if provided
```
Comment (jaimemarijke) [inline osmo_jupyter/ysi_test.py]:
```suggestion
        # Only override the "interpolation_method" kwarg if provided
```

**Review by jaimemarijke — APPROVED** [review]

File: osmo_jupyter/ysi_test.py (hunk 1 above)
Comment (jason-curtis) [reply osmo_jupyter/ysi_test.py:97]: seems like overkill - I never expect anyone to pass 'slinear'. also I don't want to go too far towards testing the functionality of pandas interpolate and think I may actually be doing that here.

**Pull request merged**